### PR TITLE
feat(issue_platform): Enable UI for Issue Platform deletion

### PR DESCRIPTION
**Pull request by armenzg**
This enables the UI for Issue Platform deletion.

This depends on #77794 being merged first.

image

### CODE REVIEW

File: static/app/views/issueList/actions/index.spec.tsx [hunk 1]
```diff
@@ -466,6 +466,32 @@ describe('IssueListActions', function () {
...
+
+      render(<WrappedComponent />, {organization: orgWithPerformanceIssuesAndDeletionUI});
+
+      // Open overflow menu
+      await userEvent.click(screen.getByRole('button', {name: 'More issue actions'}));
+
+      // Delete should be enabled
+      expect(screen.getByRole('menuitemradio', {name: 'Delete'})).not.toHaveAttribute(
+        'aria-disabled'
+      );
```
Comment (scttcper) [inline static/app/views/issueList/actions/index.spec.tsx:492]:
```suggestion
      expect(screen.getByRole('menuitemradio', {name: 'Delete'})).toBeEnabled();
```

File: static/app/views/issueDetails/actions/index.spec.tsx [hunk 2]
```diff
@@ -177,6 +177,34 @@ describe('GroupActions', function () {
...
+      {organization: org}
+    );
+
+    await userEvent.click(screen.getByLabelText('More Actions'));
+    expect(
+      await screen.findByRole('menuitemradio', {name: 'Delete'})
+    ).not.toHaveAttribute('aria-disabled');
+    expect(
+      await screen.findByRole('menuitemradio', {name: 'Delete and discard future events'})
+    ).toHaveAttribute('aria-disabled');
```
Comment (armenzg) [inline static/app/views/issueDetails/actions/index.spec.tsx:205]: I'm confused as to why this does not pass. The test runner does not give me info as to what attributes it has.
image

image

File: static/app/views/issueDetails/actions/index.spec.tsx [hunk 3]
```diff
@@ -177,6 +177,34 @@ describe('GroupActions', function () {
...
+    const org = OrganizationFixture({
+      ...organization,
+      access: [...organization.access, 'event:admin'],
+      features: [...organization.features, 'issue-platform-deletions-ui'],
+    });
+    render(
+      <Fragment>
+        <GlobalModal />
+        <GroupActions
+          group={group}
```
Comment (roggenkemper) [inline static/app/views/issueDetails/actions/index.spec.tsx:190]: group here i think is an error

File: static/app/views/issueDetails/actions/index.spec.tsx [hunk 4]
```diff
@@ -177,6 +177,34 @@ describe('GroupActions', function () {
     });
   });
 
+  it('delete for issue platform is enabled', async () => {
+    const org = OrganizationFixture({
+      ...organization,
+      access: [...organization.access, 'event:admin'],
+      features: [...organization.features, 'issue-platform-deletions-ui'],
```
Comment (roggenkemper) [inline static/app/views/issueDetails/actions/index.spec.tsx:184]: here you use `issue-platform-deletions-ui` while in the code you use `issue-platform-deletion-ui`

File: static/app/views/issueDetails/actions/index.spec.tsx [hunk 5]
```diff
@@ -209,48 +215,99 @@ describe('GroupActions', function () {
     expect(updateMock).toHaveBeenCalled();
   });
 
-  it('opens delete confirm modal from more actions dropdown', async () => {
```
Comment (armenzg) [inline static/app/views/issueDetails/actions/index.spec.tsx:212]: I'm just moving this test within the `describe` without changing it.

File: static/app/views/issueDetails/actions/index.spec.tsx [hunk 6]
```diff
@@ -209,48 +215,99 @@ describe('GroupActions', function () {
...
-      url: `/projects/${org.slug}/${project.slug}/issues/`,
-      method: 'PUT',
-      body: {},
-    });
-    const deleteMock = MockApiClient.addMockResponse({
-      url: `/projects/${org.slug}/${project.slug}/issues/`,
-      method: 'DELETE',
-      body: {},
+  describe('delete', function () {
+    it('opens delete confirm modal from more actions dropdown', async () => {
```
Comment (armenzg) [inline static/app/views/issueDetails/actions/index.spec.tsx:219]: You can ignore this test. It's just indented.

File: static/app/views/issueDetails/actions/index.spec.tsx [hunk 7]
```diff
@@ -209,48 +215,99 @@ describe('GroupActions', function () {
...
+      expect(browserHistory.push).toHaveBeenCalledWith({
+        pathname: `/organizations/${org.slug}/issues/`,
+        query: {project: project.id},
+      });
     });
-    render(
-      <Fragment>
-        <GlobalModal />
+
+    it('delete for issue platform', async () => {
```
Comment (armenzg) [inline static/app/views/issueDetails/actions/index.spec.tsx:264]: The case without the feature flag.

File: static/app/views/issueDetails/actions/index.spec.tsx [hunk 8]
```diff
@@ -209,48 +215,99 @@ describe('GroupActions', function () {
...
+      expect(await screen.findByTestId('delete-issue')).toHaveAttribute(
+        'aria-disabled',
+        'true'
+      );
+      expect(await screen.findByTestId('delete-and-discard')).toHaveAttribute(
+        'aria-disabled',
+        'true'
+      );
+    });
+    it('delete for issue platform is enabled with feature flag', async () => {
```
Comment (armenzg) [inline static/app/views/issueDetails/actions/index.spec.tsx:288]: The case with the feature flag.

**Review by roggenkemper — APPROVED** [review]

**Pull request merged**